project/profiles/capacity4more/modules/c4m/content/c4m_content_photo/c4m_content_photo.module

```diff
@@ -14,16 +14,74 @@ function c4m_content_photo_theme_registry_alter(&$theme_registry) {
 }
 
 /**
+ * Implements hook_node_presave().
+ *
+ * If a photo was uploaded and if is part of a photo album rename Album name
+ * along with a numerical suffix.
+ */
+function c4m_content_photo_node_presave($node) {
+  if ($node->type != 'photo') {
+    // Node is not a photo.
+    return;
+  }
+
+  if (!empty($node->nid)) {
+    // Node is already saved.
+    return;
+  }
+
+  $wrapper = entity_metadata_wrapper('node', $node);
+
+  if (!$wrapper->node_gallery_ref_1->getIdentifier()) {
+    // Photo is not part of an album.
+    return;
+  }
+
+  $source = file_load($wrapper->c4m_media->value()['fid']);
+  $gallery_title = $wrapper->node_gallery_ref_1->label();
+  $destination = drupal_dirname($source->uri) . '/' . transliteration_clean_filename($gallery_title) . '.' . pathinfo($source->uri, PATHINFO_EXTENSION);
```

Comment:
`format_string()` was removed by @amitaibu as we do not need to prepare the string for HTML display.